### DISCUSSION TITLE
2024-03-20

### DISCUSSION THREAD

**Discussion by yutingzhao1991** [ant-design/ant-design-web3#649] [中文社区双周会]
上次双周会记录：https://github.com/ant-design/ant-design-web3/discussions/593#discussioncomment-8662538

**Comment by yutingzhao1991** [comment]
一些新的需求：
- 添加连接状态样式 https://github.com/ant-design/ant-design-web3/issues/640
- 新组件 CryptoInput https://github.com/ant-design/ant-design-web3/discussions/638

**Comment by yutingzhao1991** [comment]
支持比特币 https://github.com/ant-design/ant-design-web3/issues/395  @AmAzing129

**Comment by yutingzhao1991** [comment]
讨论下要不要开发 web3.js 和 ethers 的适配器。

**Reply by jeasonstudio** [reply]
较真儿来说应该是三个：web3.js、ethers v5 和 ethers v6，目前 ethers v5 的用户量还很大。如果要支持 ethers，需要看下是用 ethers 单独实现下适配器，还是用 wagmi 的兼容方案：https://wagmi.sh/core/guides/ethers

**Reply by yutingzhao1991** [reply]
web3.js 也有类似的方案，只是还没有正式放到官网 https://github.com/wevm/wagmi/pull/3624/files

另外研究了一下数据，其中 NPM 下载量：
- wagmi 171026
- viem 437248
- web3 465719
- ethers 1135826

star 数：
<img width="923" alt="image" src="https://github.com/ant-design/ant-design-web3/assets/1061968/bcd03b6a-1ec3-4b76-b68e-3c42bde7fa81">

感觉不管是下载量还是 star 数的新增，web3 和 ethers 都不差。提供文档和 demo，用适配 wagmi 的方式，吸引那些用了 web3 和 ethers 的开发，这样是成本最低的，不过这样其实应该只对那些本来就想要迁移到 wagmi 的开发有吸引力。对于那些可能就想用 web3，不想用 wagmi 的估计没啥吸引力。

不过有一个问题是，大家会如何选择用什么 sdk？以及其他类似的连接钱包的组件对于其他 sdk 支持得如何？

**Reply by yutingzhao1991** [reply]
[ConnectKit](https://docs.family.co/connectkit) 和 [RainbowKit](https://www.rainbowkit.com/) 不支持 ethers，[Web3Modal](https://web3modal.com/) 是支持的。不过这几个都是从 wagmi 官网的推荐里面找到的，更多其它的就不确定了。

**Reply by jeasonstudio** [reply]
这个图比较直观：https://npmtrends.com/ethers-vs-viem-vs-wagmi-vs-web3

![screenshot-Eb9QuO9P@2x](https://github.com/ant-design/ant-design-web3/assets/17971291/9f1eda54-27e6-44dd-9019-1c407d92f5a3)

我们肯定是没资格选择用户的，这么看还是原生支持了 ethers 和 web3js 比较好，毕竟目前它俩还是大头，用户选择他们也是因为用户基数大吧。不过 wagmi/viem 还是值得投资的，未来有机会取代它们。

**Reply by jeasonstudio** [reply]
挑 ethers v6 原生实现一下，其他的先用兼容方案 @jeasonstudio

**Comment by jeasonstudio** [comment]
周会上会进行设计评审和走查问题对齐。cc @文卿

**Reply by yutingzhao1991** [reply]
对设计的时候可以参考一些网站看看有没有需要借鉴的：
- https://wagmi.sh/
- https://www.rainbowkit.com/zh-CN
- https://family.co/
- https://web3js.readthedocs.io/en/v1.10.0/#
- https://walletconnect.com/
- https://www.dynamic.xyz/

**Reply by yutingzhao1991** [reply]
感觉不同的框架有不同的特质，官网也要能体现自己的特质，能够吸引适合的用户来使用，从这个角度看我觉得 Ant Design Web3 要能够体现出：

- 高质量的 UI 组件，基于 Ant Design 强大的保障。包括交互体验、测试覆盖率、UI 细节等。
- 丰富的组件，不仅仅是 ConnectModal，能够解决开发的问题，提升开发效率
- 定制主题，随心所欲（这个也是 antd 的宣传语）

**Reply by yutingzhao1991** [reply]
<img width="1472" alt="image" src="https://github.com/ant-design/ant-design-web3/assets/1061968/99960eb7-3284-43b3-88e6-b50ec944ba28">

antd 首页的这个主题切换体验起来还挺酷的

**Comment by jeasonstudio** [comment]
遗留 PR 推进：
* https://github.com/ant-design/ant-design-web3/pull/387
* https://github.com/ant-design/ant-design-web3/pull/532
* https://github.com/ant-design/ant-design-web3/pull/631
* https://github.com/ant-design/ant-design-web3/pull/656

**Comment by yutingzhao1991** [comment]
讨论看看要不要加一个【Why Ant Design Web3？】的文档，列出来我们的优点，用表格列出来支持的特性。

**Reply by jeasonstudio** [reply]
加在这个页面下方：https://web3.ant.design/guide/ant-design-web3 只列 features，暂时不跟别的 lib 对比。 @yutingzhao1991
